src/app.py

```diff
@@ -854,15 +876,24 @@ def publish_datastage(identifier):
                 logger.info(identifier + "\t" + dataset_uuid + "\tNEO4J-update-ancestors\t" + update_q)
                 neo_session.run(update_q)
                     
+            # verify that dataset status has been successfully updated to "Published"
+            published_q = "match (e:Entity {uuid:'" + dataset_uuid + "'}) RETURN e.status as status"
+            rval = neo_session.run(published_q).data()
+            if rval[0]['status'].upper() != "PUBLISHED":
+                return jsonify({"error": f"Internal error. Dataset with id {dataset_uuid} status not successfully changed to 'Published'"}), 500
+
 
             # DOI gets generated here
-            # TODO: Enable the commented code once the integration work with pipeline has been completed for story #354.
             # Note: moved dataset title auto generation to entity-api - Zhou 9/29/2021
-            # nexus_token = app_manager.nexus_token_from_request_headers(request.headers)
-            # datacite_doi_helper = DataCiteDoiHelper()
-            # datacite_doi_helper.create_dataset_draft_doi(node)
-            # # This will make the draft DOI created above 'findable'....
-            # datacite_doi_helper.move_doi_state_from_draft_to_findable(node, nexus_token)
+            auth_tokens = auth_helper.getAuthorizationTokens(request.headers)
+            datacite_doi_helper = DataCiteDoiHelper()
+
+            entity_instance = EntitySdk(token=auth_tokens, service_url=app.config['ENTITY_WEBSERVICE_URL'])
+            entity = entity_instance.get_entity_by_id(dataset_uuid)
+            entity_dict = vars(entity)
+            datacite_doi_helper.create_dataset_draft_doi(entity_dict)
```

Comment:
We should add try/except blocks to both the draft DOI creation and the state change to findable calls. In practice, we only update the state once the draft DOI gets created successfully, and should exit and throw errors back if there's an exception. Same for the state change call, if something doesn't go right, we exit and throw error back so the DOI creator (AKA, @shirey ) would manually fix the issue.